Packs/qualys/Integrations/Qualysv2/Qualysv2.py

```diff
@@ -3480,14 +3493,26 @@ def main():  # pragma: no cover
         elif command == 'fetch-assets':
             last_run = demisto.getAssetsLastRun()
             demisto.debug(f'saved lastrun assets: {last_run}')
+
             fetch_stage = last_run.get('stage', 'assets')
 
             if fetch_stage == 'assets':
                 start_time = time.time()
                 demisto.debug(f'Starting fetch for assets, {start_time=}')
                 assets, new_last_run, total_assets, snapshot_id = fetch_assets(client=client, assets_last_run=last_run)
                 check_fetch_duration_time(start_time, last_run.get('limit', HOST_LIMIT))
-
+                # If assets, create list of QIDs and deduplicate later.
+                if assets:
+                    qid_list = []
+                    for asset in assets:
+                        if asset.get('DETECTION', {}).get('QID'):
+                            qid_list.append(asset.get('DETECTION', {}).get('QID'))
+                # Look for QIDs already in last_run and deduplicate.
+                if qid_list:
+                    if qids_last_run := last_run.get('qids'):
+                        qid_list.extend(qids_last_run)
+                    qid_set = set(qid_list)
+                    new_last_run['qids'] = list(qid_set)
```

Comment:
`qid_list` is not defined in case no `assets` exist
```suggestion
                if assets:
                    qid_list = []
                    for asset in assets:
                        if asset.get('DETECTION', {}).get('QID'):
                            qid_list.append(asset.get('DETECTION', {}).get('QID'))
                    # Look for QIDs already in last_run and deduplicate.
                    if qid_list:
                        if qids_last_run := last_run.get('qids'):
                            qid_list.extend(qids_last_run)
                        qid_set = set(qid_list)
                        new_last_run['qids'] = list(qid_set)
```